recipes/nest-desktop/meta.yaml

```
@@ -0,0 +1,45 @@
+{% set name = "nest-desktop" %}
+{% set version = "3.1.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/nest-desktop-{{ version }}.tar.gz
+  sha256: 49051aeefc8be189ae7dfb64ad983b7a0a5636765a79febd3347194b57ba61ba
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  noarch: python
+  entry_points:
+    - nest_desktop = nest_desktop.app:run
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - python >=3.6
+
+test:
+  imports:
+    - nest_desktop
+  commands:
+    - pip check
```

```suggestion
  commands:
    - pip check
    - nest_desktop --help
```
Normally, we should also test that the CLI is installed correctly, but if we start your app, it doesn't return and must be force quit?

---

The bash script `nest-desktop` shows help when no sub-command is used:

```
~$ nest-desktop
NEST Desktop 3.1.2
------------------
Usage: nest-desktop status|start|stop|restart [-h <HOST>] [-p <PORT>]

Commands:
  status      display the status of NEST Desktop
  start       start a new server instance for NEST Desktop
  stop        stop a server instance running on <HOST>:<PORT>
  restart     restart (i.e. stop and start) a server on <HOST>:<PORT>

Options:
  -h <HOST>   use hostname/IP address <HOST> for the server [default: 127.0.0.1]
  -p <PORT>   use port <PORT> for opening the socket [default: 8000]
~$
```

To answer your question: Yes, it has to be force quit since it is served forever. (Press CRTL + C to stop.)

We could add an argument to test it without running forever:
```
def run(host=NEST_DESKTOP_HOST, port=NEST_DESKTOP_PORT, test=False):

    web_dir = os.path.join(os.path.dirname(__file__), "app")
    os.chdir(web_dir)

    Handler = http.server.SimpleHTTPRequestHandler
    
    if test:
        socketserver.TCPServer((host, port), Handler)
        print('Test successful.')
    else:
        with socketserver.TCPServer((host, port), Handler) as httpd:
            httpd.serve_forever()
```

What do you think?

---

If the bash script returns after printing the help, then use the suggestion above. If not, then I don't know how to kill your app after it has been started. Perhaps:

```suggestion
  commands:
    - pip check
    - nest-desktop start  # [unix]
    - nest-desktop stop   # [unix]
```

These tests are for testing that the packaging was done correctly, so we don't need to test your app, just test whether the app is installed into the conda environment correctly. i.e. that the bash script is found on the path.

I think it would be helpful for Windows users (who cannot use the bash script) to add the help when launching the app from python directly, but this is not required.